zerver/lib/test_helpers.py

```diff
@@ -598,3 +601,98 @@ def zulip_reaction_info() -> Dict[str, str]:
         emoji_code='zulip',
         reaction_type='zulip_extra_emoji',
     )
+
+class SpamInTestOutputException(Exception):
+    pass
+
+class SpamFinder:
+    def __init__(self) -> None:
+        self.latest_test_name = ""
+        valid_line_patterns = [
+            "^Running ",  # Example: Running zerver.tests.test_attachments.AttachmentsTests.test_delete_unauthenticated
+            "^\\*\\* Test is TOO slow: ",  # Example: ** Test is TOO slow: analytics.tests.test_counts.TestRealmActiveHumans.test_end_to_end (0.581 s)
+            "^----------------------------------------------------------------------",  # Example: ----------------------------------------------------------------------
+            "^INFO: URL coverage report is in",  # Example: INFO: URL coverage report is in var/url_coverage.txt
+            "^INFO: Try running:",  # Example: INFO: Try running: ./tools/create-test-api-docs
+            "^-- Running tests in",  # Example: -- Running tests in parallel mode with 4 processes
+            "^OK",  # Example: OK
+            "^Ran [0-9]+ tests in",  # Example: Ran 2139 tests in 115.659s
+            "^Destroying test database for alias ",  # Destroying test database for alias 'default'...
+            "^Using existing clone",
+            "^\\*\\* Skipping ",
```

Comment:
Let's put the comments on the line before the expression, rather than the same line; it'll be more readable that way.

We may want to do a bit of work to remove some of these output lines from the `test-backend` default output as a follow-up project.  E.g. @showell is the "URL coverage report" something we should just delete at this point, now that we have more powerful tools for checking that we've tested all our API endpoints?

And I feel like the "Destroying test database for alias" type output could be hidden behind a verbose flag.